internal/appsec/dyngo/instrumentation/httpsec/http.go

```diff
@@ -42,18 +44,36 @@ type (
 		// Status corresponds to the address `server.response.status`.
 		Status int
 	}
+
+	// SDKBodyOperationArgs is the SDK body operation arguments.
+	SDKBodyOperationArgs struct {
+		Body interface{}
+	}
+
+	// SDKBodyOperationRes is the SDK body operation results.
+	SDKBodyOperationRes struct {
+	}
 )
 
+// MonitorParsedBody starts and finishes the SDK body operation.
+func MonitorParsedBody(ctx context.Context, body interface{}) {
+	if parent := fromContext(ctx); parent != nil {
+		op := StartSDKBodyOperation(parent, SDKBodyOperationArgs{Body: body})
+		op.Finish()
+	} else {
+		log.Debug("appsec: no valid parent HTTP context upon SDKBody operation start" +
+			" AppSec may be disabled")
```

Comment:
Under the new usage assumptions of calling this function only when appsec is enabled, we can now better log a legitimate error message:
```suggestion
		log.Error("appsec: parsed http body monitoring ignored: could not find the http handler instrumentation metadata in the request context: the request handler is not being monitored by a middleware function or the given context is not the expected request context")
```